CMakeLists.txt

```
@@ -40,11 +40,13 @@ endif()
 set(CMAKE_CXX_STANDARD 17 CACHE STRING "The C++ standard whose features are requested to build this target.")
 set(CMAKE_C_STANDARD   11 CACHE STRING "The C standard whose features are requested to build this target.")
 
-if(DEFINED GLIBCXX_USE_CXX11_ABI)
-  if(${GLIBCXX_USE_CXX11_ABI} EQUAL 1)
-    set(CXX_STANDARD_REQUIRED ON)
-    string(APPEND CMAKE_CXX_FLAGS " -D_GLIBCXX_USE_CXX11_ABI=1")
-  else()
+include(cmake/CheckAbi.cmake)
+string(APPEND CMAKE_CXX_FLAGS " -D_GLIBCXX_USE_CXX11_ABI=${GLIBCXX_USE_CXX11_ABI}")
+if(${GLIBCXX_USE_CXX11_ABI} EQUAL 1)
+  set(CXX_STANDARD_REQUIRED ON)
+else()
+  if(${CMAKE_CXX_COMPILER} MATCHES "g\\+\\+$" AND NOT ${CMAKE_CXX_COMPILER} MATCHES "clang\\+\\+$")
+    # clang doesn't recognize this flag
     # Please note this is required in order to ensure compatibility between gcc 9 and gcc 7
     # This could be removed when all Linux PyTorch binary builds are compiled by the same toolchain again
     string(APPEND CMAKE_CXX_FLAGS " -fabi-version=11")
```

Wouldn't that be better?
```suggestion
    append_cxx_flag_if_supported("-fabi-version=11" CMAKE_CXX_FLAGS)
```

---

clang doesn't seem to support "-fabi-version" flag. Will `append_cxx_flag_if_supported` apply this flag on g++ only?